### PR TITLE
Docksal RC Version

### PR DESCRIPTION
**Pull request by sean-e-dietrich**
Added ability for users to select using the RC/Latest version of docksal by setting the `DOCKSAL_USE_RC=1` variable within their `$HOME/.docksal/docksal.env`

This assumes that we tag develop branch with `v1.10.0-rc1` this doesn't necessarily check to see if we using the `-rcX` versioning but just grabs the latest "release" within github which corresponds with the latest tag.

Solves https://github.com/docksal/docksal/issues/673

### CODE REVIEW

File: bin/fin [hunk 1]
```diff
@@ -208,24 +208,29 @@ PATH="$CONFIG_BIN_DIR:$PATH"
...
-URL_REPO_GRAV="https://github.com/docksal/example-grav.git"
-URL_REPO_GATSBY="https://github.com/docksal/example-gatsby.git"
-URL_REPO_LARAVEL="https://github.com/docksal/example-laravel.git"
-URL_REPO_HUGO="https://github.com/docksal/example-hugo.git"
-URL_REPO_SYMFONY_SKELETON="https://github.com/docksal/example-symfony-skeleton.git"
-URL_REPO_SYMFONY_WEBAPP="https://github.com/docksal/example-symfony-webapp.git"
-URL_REPO_BACKDROP="https://github.com/docksal/example-backdrop.git"
-URL_ADDONS_HOSTING="https://raw.githubusercontent.com"
-URL_ADDONS_REPO="$URL_ADDONS_HOSTING/docksal/addons"
+REPO_OWNER="docksal"
```
Comment (achekulaev) [inline bin/fin]: What is the purpose of this constant? It's not like we are going to change it abruptly :)

File: bin/fin [hunk 2]
```diff
@@ -4264,6 +4276,29 @@ update ()
...
+# Return latest tag's commit url
+get_latest_release ()
+{
+	# Get the latest release tag
+	URL_REPO_RELEASE="${GITHUB_API}/repos/${REPO_URL}/releases/latest"
+	local latest_tag=$(curl -sSL $URL_REPO_RELEASE | grep tag_name | sed 's/\"tag_name": \"\(.*\)",/\1/' | tr -d '[:space:]')
+
+	# Get information about that tag as we need to get the sha tag associated with that
+	URL_REPO_TAG="${GITHUB_API}/repos/${REPO_URL}/git/refs/tags/${latest_tag}"
+	local commit_sha=$(curl -sSL $URL_REPO_TAG | grep sha | sed 's/\"sha": \"\(.*\)",/\1/' | tr -d '[:space:]')
```
Comment (achekulaev) [inline bin/fin]: Why is there a need to get commit? You can just use tag name

```bash
https://raw.githubusercontent.com/docksal/docksal/${latest_tag}/bin/fin
```

Comment (sean-e-dietrich) [reply bin/fin]: @achekulaev documentation suggested I needed to get the SHA in order to get the file

File: bin/fin (hunk 1 above)
Comment (achekulaev) [reply bin/fin]: Can we please revert these the changes unrelated to Docksal RC Version for now?

File: bin/fin [hunk 3]
```diff
@@ -4264,6 +4276,25 @@ update ()
...
+	local latest_tag=$(curl -sSL $URL_REPO_RELEASE | grep tag_name | sed 's/\"tag_name": \"\(.*\)",/\1/' | tr -d '[:space:]')
+
+	# Return the URL associated with that sha hash to get that specific version of fin's id
+	URL_REPO_COMMIT_URL="${URL_REPO}/${latest_tag}/bin/fin"
+	echo "${URL_REPO_COMMIT_URL}"
+}
+
+# Return the url for fin. If person has set DOCKSAL_USE_RC then we return that otherwise return masters URL
+# which will return the latest stable version.
+get_latest_version_fin ()
```
Comment (achekulaev) [inline bin/fin]: > Return the url for fin.

`get_latest_fin_url` ?

File: bin/fin [hunk 4]
```diff
@@ -6377,8 +6414,9 @@ case "$1" in
 		elif [[ "$1" == "--project-images" ]]; then
 			update_project_images
 		elif [[ "$1" == "--self" ]]; then
-			echo "Downloading $URL_FIN"
-			curl -kfsSL "$URL_FIN?r=$RANDOM" | sudo tee "$FIN_PATH_UPDATED" >/dev/null
+			fin_file_url=$(get_latest_version_fin)
```
Comment (achekulaev) [inline bin/fin]: The variable where `get_latest_version_fin` result is stored is called `fin_url` in the `check_for_updates` function, for consistency it should be called the same here.

**Pull request merged**